cmake/configure/configure_20_trilinos.cmake

```diff
@@ -272,9 +272,45 @@ macro(feature_trilinos_find_external var)
       else()
         message(
           STATUS
-          "Tpetra was found but is not usable! Disabling Tpetra support."
+          "Tpetra was found but is not usable! Disabling Tpetra support and searching for common errors."
           )
         set(TRILINOS_WITH_TPETRA OFF)
+
+        check_cxx_symbol_exists(HAVE_TPETRA_INT_LONG_LONG "TpetraCore_config.h" DEAL_II_HAVE_TPETRA_INT_LONG_LONG)
+        check_cxx_symbol_exists(HAVE_TPETRA_INT_INT "TpetraCore_config.h" DEAL_II_HAVE_TPETRA_INT_INT)
+
+        if(DEAL_II_HAVE_TPETRA_INT_INT AND DEAL_II_WITH_64BIT_INDICES)
+          message(
+            STATUS 
+            "  Tpetra is installed with 32 bit global indices but you want to build deal.II with 64. "
+          )
+          message(
+            STATUS 
+            "  Either rebuild Trilinos with -DTPETRA_INST_INT_INT=OFF and -DTPETRA_INST_LONG_LONG=ON "
```

Comment:
```suggestion
            "  Either rebuild Trilinos with -DTpetra_INST_INT_LONG_LONG=ON "
```

Reply:
Ah, you're right.

Reply:
I also added a check for the option of `long` as global ordinate type,
such that the message tells the user to switch off the instantiation that is currently on.